### ISSUE TITLE
Certificates warning messages in Fabric with BFT orderers

### ISSUE THREAD

**Issue by satota2** [hyperledger/fabric#4404]
## Description
I have currently been trying to run BFT orderer using the BFT samples on fabric-samples.
When the samples are run, the following Certificates warning messages are probably periodically output in each orderer's log:
```
2023-08-23 09:22:05.892 UTC 0dc2 WARN [orderer.common.cluster.step] checkExpiration -> Certificate of orderer2.example.com from orderer2.example.com:7052 for channel mychannel expires in less than -2562047h47m16.854775808s2
```
It seems that the initial value is set in exp.expiresAt.
Is this expected behavior?

### Steps to reproduce

I have used the following versions.
- fabric (Commit SHA: 967734401 (the latest in main branch))
- fabric-samples:
  - BFT samples I used:
    - test-network
    - test-network-nano
  - Memo:
    - We reproduced it in both samples.
    - Fabric network with RAFT orderer did not seem to output this warning.

**Comment by yacovm** [comment]
>  Is this expected behavior?

Of course not.

Below is a git patch that fixes this. Please feel free to do a pull request.

```
diff --git a/orderer/common/cluster/remotecontext.go b/orderer/common/cluster/remotecontext.go
index 728409d0b..bd271869e 100644
--- a/orderer/common/cluster/remotecontext.go
+++ b/orderer/common/cluster/remotecontext.go
@@ -14,6 +14,7 @@ import (
 
        "github.com/hyperledger/fabric-protos-go/orderer"
        "github.com/hyperledger/fabric/common/flogging"
+       "github.com/hyperledger/fabric/common/util"
        "github.com/pkg/errors"
        "go.uber.org/zap"
        "google.golang.org/grpc"
@@ -101,7 +102,6 @@ func (rc *RemoteContext) NewStream(timeout time.Duration) (*Stream, error) {
        s.expCheck = &certificateExpirationCheck{
                minimumExpirationWarningInterval: rc.minimumExpirationWarningInterval,
                expirationWarningThreshold:       rc.certExpWarningThreshold,
-               expiresAt:                        rc.expiresAt,
                endpoint:                         s.Endpoint,
                nodeName:                         s.NodeName,
                alert: func(template string, args ...interface{}) {
@@ -109,6 +109,10 @@ func (rc *RemoteContext) NewStream(timeout time.Duration) (*Stream, error) {
                },
        }
 
+       if cert := util.ExtractCertificateFromContext(stream.Context()); cert != nil {
+               s.expCheck.expiresAt = cert.NotAfter
+       }
+
        err = stream.Auth()
        if err != nil {
                return nil, errors.Wrap(err, "failed to create new stream")
diff --git a/orderer/common/cluster/util.go b/orderer/common/cluster/util.go
index e89c7a188..d5c60b5cc 100644
--- a/orderer/common/cluster/util.go
+++ b/orderer/common/cluster/util.go
@@ -682,7 +682,7 @@ type certificateExpirationCheck struct {
 
 func (exp *certificateExpirationCheck) checkExpiration(currentTime time.Time, channel string) {
        timeLeft := exp.expiresAt.Sub(currentTime)
-       if timeLeft > exp.expirationWarningThreshold {
+       if exp.expiresAt.IsZero() || timeLeft > exp.expirationWarningThreshold {
                return
        }
 

```

**Comment by satota2** [comment]
@yacovm 
> Below is a git patch that fixes this. Please feel free to do a pull request.

Thank you for you suggesting the fix.
We have confirmed that the warning messages are no longer output with the fix.

Based on your suggestion, I created and submitted the following PR.
https://github.com/hyperledger/fabric/pull/4411

**Comment by satota2** [comment]
This Issue is closed because it has been resolved by the PR[#4411](https://github.com/hyperledger/fabric/pull/4411).
